docs/guides/javascript.rst

```diff
@@ -542,6 +542,65 @@ Plugins that load a widget layout via Ajax should initialize via this module:
        widgets.init();
    });
 
+Module ``elgg/lightbox``
+------------------------
+
+Elgg is distributed with the Colorbox jQuery library. Please go to http://www.jacklmoore.com/colorbox for more information on the options of this lightbox.
+
+Use ``elgg-lightbox`` class on your anchor element to bind it to a lightbox.
+
+.. note:: For performance reasons lightbox bindings are not performed automatically. You need to use ``elgg_load_js('lightbox')`` and ``elgg_load_css('lightbox')`` to load required assets.
+
+You may apply colorbox options to an individual ``elgg-lightbox`` element by setting the attribute ``data-colorbox-opts`` to a JSON settings object.
+
+.. code:: php
+
+   echo elgg_view('output/url', [
+      'text' => 'Open lightbox',
+      'href' => 'ajax/view/my_view',
+      'class' => 'elgg-lightbox',
+      'data-colorbox-opts' => json_encode([
+         'width' => '300px',
+      ])
+   ]);
+
+For images, use ``elgg-lightbox-photo`` class, or pass ``{"photo": true}`` with the ``data-colorbox-opts``.
+
+To support a hidden ``div`` as the html source of the lightbox, add ``{"inline": true}`` to your ``data-colorbox-opts``.
+
+Use ``"getOptions", "ui.lightbox"`` plugin hook to filter options passed to ``$.colorbox()`` whenever a lightbox is opened.
+
+``elgg/lightbox`` AMD module should be used to open and close the lightbox programmatically:
+
+.. code:: js
+
+   define(function(require) {
+      var lightbox = require('elgg/lightbox');
+      var spinner = require('elgg/spinner');
+
+      lightbox.open({
+         html: '<p>Hello world!</p>',
+         onClosed: function() {
+            lightbox.open({
+               onLoad: spinner.start,
+               onComplete: spinner.stop,
+               photo: true,
+               href: 'https://elgg.org/cache/1457904417/default/community_theme/graphics/logo.png',
+            });
+         }
+      });
+   });
+
+.. note:: At this time, you should not rely on lightbox CSS to be loaded when you require the lightbox module. Use ``elgg_load_css('lightbox')``.
+
+To support gallery sets (via ``rel`` attribute), you need to bind colorbox to a specific selector:
+
+..code:: js
+
+   require(['elgg/lightbox'], function() {
+      $('.my-gallery').colorbox({photo: true});
```

Comment:
We should add a lightbox "bind" method so we don't instruct users to call colorbox API directly.